core/modules/search/search.admin.inc

```diff
@@ -79,6 +79,16 @@ function search_admin_settings($form, &$form_state) {
     '#options' => $module_options,
     '#description' => t('Only one type of item will be searched by default. This selection will determine what appears at http://www.example.com/search.')
   );
+  $form['logging'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Logging')
+  );
+  $form['logging']['search_logging'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Log searches'),
+    '#default_value' => $config->get('search_logging'),
+    '#description' => t('If checked, all searches will be logged. Uncheck to skip logging. Logging may affect performance.'),
+  );
```

Comment:
It doesn't look like this would get saved to the config file on form submit. Let's test this manually to confirm.